mux/option.go

```diff
@@ -34,7 +31,7 @@ func WithGRPC(server *grpc.Server) Option {
 
 // WithGracePeriod sets the wait duration for graceful shutdown.
 func WithGracePeriod(d time.Duration) Option {
```

Comment:
how about using `WithGracePeriod` with optional parameter. `WithGracePeriod(0)` says with 0 second as grace period. like:
```go
WithGracePeriod(optionalDuration ...time.Duration) {
    var d time.Duration
    if len(optionalDuration) == 0 {
         d = optionalDuration[0]
    }
}
```

Reply:
option parameters ike `WithGracePeriod` are for when you want to *override* some default. Having an option `WithGracePeriod()` that doesn't actually take an override value isn't really needed right?

Doing `if d==0 { d = default }` is almost the same, but it's only being done as a safety measure (because using <=0 value in timer can cause panics)..

Reply:
> because using <=0 value in timer can cause panics

can you explain why ?

Reply:
https://pkg.go.dev/time#NewTicker

It's how an internal runtime timer behaves in Go. It panics for 0 or negative values. I have not explicitly tried it with context.WithTimeout though.

Reply:
then lets just use this instead of `if d <= 0` instead of `if d == 0` ?

Reply:
oh yes, that's a mistake. fixed it.